src/components/user-events/calendar-heatmap.tsx

```diff
@@ -0,0 +1,86 @@
+"use client";
+
+import {
+  addDays,
+  addWeeks,
+  endOfDay,
+  isBefore,
+  isSameDay,
+  isSameMonth,
+  previousMonday,
+  startOfMonth,
+} from "date-fns";
+import { Box, HStack, VStack } from "styled-system/jsx";
+
+const WEEK_DAYS: number[] = [0, 1, 2, 3, 4, 5, 6];
+const MONTH_WEEKS: number[] = [0, 1, 2, 3, 4, 5];
+const TILE_COLORS = {
+  missing: {
+    base: "colorPalette.1/5",
+    _dark: "colorPalette.4/5",
+  },
+  inactive: {
+    base: "colorPalette.1",
+    _dark: "colorPalette.6",
+  },
+  active: {
+    base: "colorPalette.7",
+    _dark: "colorPalette.10",
+  },
+};
+
+type CalendarHeatmapProps = {
+  lastDate: Date;
+  activeDates: Date[];
+};
+
+const getDayColor = (day: Date, lastDate: Date, activeDates: Date[]) => {
+  const isDateActive =
+    activeDates.find((activeDate) => {
+      return isSameDay(day, activeDate);
+    }) !== undefined;
+  const isInMonth =
+    isBefore(day, endOfDay(lastDate)) && isSameMonth(day, lastDate);
```

Comment:
This seems redundant to me; isn't `isInMonth` and `isSameMonth` technically the same?

Either use this:
```suggestion
  const isInMonth = isSameMonth(day, lastDate);
```
Or just get rid of the `isInMonth` variable entirely.

Reply:
Not quite: the `isBefore(day, enfOfDay(lastDate))` ensures that days after `lastDate` but still in the same month don't get highlighted. This is what allows the component to display the current month **only up to the current date** and not in its entirety.